dkpro-core-castransformation-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/castransformation/Backmapper.java

```diff
@@ -145,15 +147,22 @@ public void process(final JCas aJCas)
     }
     
     private AlignedString getAlignedString(JCas aSomeCase, String from, String to)
-        throws AnalysisEngineProcessException
-    {
+            throws AnalysisEngineProcessException, CASException {
         CAS baseCas = aSomeCase.getCasImpl().getBaseCAS();
 
         // Try to get the AlignedString for the current JCas.
         AlignmentStorage asstore = AlignmentStorage.getInstance();
         AlignedString as = asstore.get(baseCas, to, from);
 
-        // If there is none we have to fail.
+        if (as == null) {
+            // Attempt to reconstruct the alignment from the SofaChangeAnnotations.
+            // This only works when they have not been altered in the mean time.
```

Comment:
I'll add some info logging output about the fallback happening going on here. Should we eventually make it possible to opt this out/in?

Reply:
The only reason I could imaging that somebody would disable the reconstruction of the AS from the CAS is to detect bugs in their setup causing the AS to be lost unintentionally. I imagine this to be a very rare case. Maybe I miss something. So I do not have a strong opinion here. If you find it useful for your case, add the parameter, otherwise we can also wait until somebody needs it.

Reply:
Let's then keep it simple until a pressing need comes up.